lock-api-objects/src/test/java/com/palantir/lock/watch/LockEventTest.java

```diff
@@ -0,0 +1,215 @@
+            ChangeMetadata.created(BYTES_CREATED));
+    private static final LockRequestMetadata LOCK_REQUEST_METADATA = LockRequestMetadata.of(ImmutableMap.of(
+            LOCK_1,
+            CHANGE_METADATA_LIST.get(0),
+            LOCK_2,
+            CHANGE_METADATA_LIST.get(1),
+            LOCK_3,
+            CHANGE_METADATA_LIST.get(2),
+            LOCK_4,
+            CHANGE_METADATA_LIST.get(3)));
+
+    // TimeLock (Server) serializes and AtlasDB (Client) deserializes.
+    // These are the respective mappers used internally by Conjure.
+    private static final ObjectMapper SERIALIZATION_MAPPER =
+            ObjectMappers.newServerObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private static final ObjectMapper DESERIALIZATION_MAPPER =
+            ObjectMappers.newClientObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    // This mapper is used to ensure that two JSONs are equal excluding indentation
+    private static final ObjectMapper VERIFYING_MAPPER = new ObjectMapper();
+
+    @Test
+    public void oldLockEventIsBaseline() {
+        assertSerializedEquals(OLD_LOCK_EVENT, "baseline");
+        assertDeserializedEquals("baseline", OLD_LOCK_EVENT, OldLockEvent.class);
+    }
+
+    @Test
+    public void serializedFormatIsUnchangedForAbsentMetadata() {
+        assertSerializedEquals(BASELINE_LOCK_EVENT, "baseline");
+        assertDeserializedEquals("baseline", BASELINE_LOCK_EVENT, LockEvent.class);
+    }
+
+    @Test
+    public void serializesMetadataIfPresent() {
+        LockEvent lockEventWithMetadata =
+                ImmutableLockEvent.copyOf(BASELINE_LOCK_EVENT).withMetadata(LOCK_REQUEST_METADATA);
+        assertSerializedEquals(lockEventWithMetadata, "baseline-with-metadata");
+        assertDeserializedEquals("baseline-with-metadata", lockEventWithMetadata, LockEvent.class);
+    }
+
+    @Test
+    public void serializesAsLockWatchEventWithMetadata() {
+        LockWatchEvent asLockWatchEvent =
+                ImmutableLockEvent.copyOf(BASELINE_LOCK_EVENT).withMetadata(LOCK_REQUEST_METADATA);
+        assertSerializedEquals(asLockWatchEvent, "baseline-with-metadata");
+        assertDeserializedEquals("baseline-with-metadata", asLockWatchEvent, LockWatchEvent.class);
+    }
+
+    @Test
+    public void serializesEmptyMetadataMap() {
+        LockEvent lockEventWithMetadata =
+                ImmutableLockEvent.copyOf(BASELINE_LOCK_EVENT).withMetadata(LockRequestMetadata.of(ImmutableMap.of()));
+        assertSerializedEquals(lockEventWithMetadata, "baseline-empty-metadata-map");
+        assertDeserializedEquals("baseline-empty-metadata-map", lockEventWithMetadata, LockEvent.class);
+    }
+
+    // This property should be guaranteed by the Conjure spec
+    @Test
+    public void oldClientIgnoresMetadata() {
+        assertDeserializedEquals("baseline-with-metadata", OLD_LOCK_EVENT, OldLockEvent.class);
+    }
+
+    @Test
+    public void serializesSparseMetadata() {
```

Comment:
I initially had this set up as a random test, but realized (via some Feedback from Sam in the other PR) that we actually jus want to test the case of sparse metadata here, i.e. not every lock is assigned metadata. For this purpose, I think a deterministic test makes more sense